### PR TITLE
kubernetes helm charts

### PR DESCRIPTION
**Pull request by rokroskar**
Produce k8s helm charts for deploying renga. 

- [x] integrate the pieces from `docker-compose.yml`
- [x] automate adding the `renga-ui` app to gitlab db
- [x] move the gitlab config from the environment variable to a template file
- [x] remove unneeded pieces from gitlab (redis, postgres)
- [x] persistent storage for gitlab configuration (#129)

---

addresses #135  and https://github.com/SwissDataScienceCenter/incubator-renga-k8s/issues/1

### CODE REVIEW

File: charts/renga/charts/gitlab/values.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  # existingClaim:
+
+  # storageClass: "-"
+  accessMode: ReadWriteOnce
+  size: 30Gi
+
+  gitlab-data:
+    subPath: data
+    mountPath: /var/opt/gitlab
+  gitlab-config:
```
Comment (rokroskar) [inline charts/renga/charts/gitlab/values.yaml]: I think the mountpaths for config and logs are switched @leafty?

File: charts/renga/values.yaml [hunk 2]
```diff
@@ -26,6 +26,8 @@ global:
     postgresDatabase: gitlabhq_production
     postgresUser: gitlab
     postgresPassword: gitlab
+    clientSecret: open_secret
+    rootPassword: root_password
```
Comment (leafty) [inline charts/renga/values.yaml]: This doesn't need to be global AFAIK.
can we move it to the `gitlab section`? Rename the exisitng `gitlab.password` value to `gitlab.rootPassword` if desired.

Comment (rokroskar) [reply charts/renga/values.yaml]: Well I think we want to provide sensible defaults in `charts/gitlab/values.yaml` and potentially override them in the top-level `values.yaml`. But you're right, these should not in any case be in a part of `global`.

Comment (rokroskar) [reply charts/renga/values.yaml]: actually I think it does need to be global because both keycloak and gitlab need to access the value?

Comment (leafty) [reply charts/renga/values.yaml]: No, that values is only found in `charts/gitlab/templates/secret.yaml`

Comment (rokroskar) [reply charts/renga/values.yaml]: ah I thought you were referring also to `clientSecret` -- yes, `rootPassword` can be removed, I just did in 97319b0

File: charts/minikube-values.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  enabled: true
+  annotations:
+    kubernetes.io/ingress.class: nginx
+    nginx.ingress.kubernetes.io/ssl-redirect: false
+  hosts:
+  - ""
+
+global:
+  renga:
+    domain: "192.168.99.100"
```
Comment (leafty) [inline charts/minikube-values.yaml]: minikube ip are not fixed

Comment (rokroskar) [reply charts/minikube-values.yaml]: fixed in 44159be

**Pull request merged**